otherlibs/unix/caml/unixsupport.h

```diff
@@ -46,7 +46,12 @@ struct filedescr {
     SOCKET socket;
   } fd;                   /* Real windows handle */
   enum { KIND_HANDLE, KIND_SOCKET } kind;
-  _Atomic int crt_fd;     /* C runtime descriptor */
+#ifdef __cplusplus
+  std::atomic<int>
+#else
+  _Atomic int
+#endif
```

Comment:
We have `atomic_intnat` in misc.h, can you use it here? Otherwise, would you maybe define `atomic_int` in misc.h?